lib/Target/CBackend/TopologicalSorter.cpp

```diff
@@ -0,0 +1,86 @@
+// Topological Sorting based on C# code from
+// http://tawani.blogspot.com/2009/02/topological-sorting-and-cyclic.html
+// which is based on Java code from
+// http://www.java2s.com/Code/Java/Collections-Data-Structure/Topologicalsorting.htm
```

Comment:
The license for this code (All rights reserved) is incompatible with this project, and thus I can't merge anything tainted by it. However, I think this sort is the same as the reverse-post-order pass in LLVM will give you.

Reply:
Ok, im sorry for this. When i have time i will look for an alternative solution